### PR TITLE
Merge JsonElementType into JsonElement

### PR DESCRIPTION
**Pull request by anhoangphuc**
Merge JsonElementType into JsonElement. Use Enum instead integer

### CODE REVIEW

File: src/main/java/woocommerce/utils/ReflectionUtils.java [hunk 1]
```diff
@@ -33,14 +33,19 @@ public static ProductVariable jsonStringToProductVariable(String json) {
...
-                            prop = Integer.parseInt((String) result.get(field.getAnnotation(JsonElement.class).key()));
+                    TypeOfJsonElement typeOfElement = field.getAnnotation(JsonElement.class).type();
+
+                    if (typeOfElement != TypeOfJsonElement.NONE) {
+                        //need to cast to specific type
+                        if (typeOfElement == TypeOfJsonElement.INTEGER) {
+                            Integer prop = Integer.parseInt((String) result.get(key));
+                            field.set(productVariable, prop);
+                        } else if (typeOfElement == TypeOfJsonElement.STRING) {
+                            String prop = result.get(key).toString();
```
Comment (weirdobeardo48) [inline src/main/java/woocommerce/utils/ReflectionUtils.java]: There could be a chance that result.get(key) return null, so toString() method would throw NullPointerException. I think in this case, cast to String is better

Comment (anhoangphuc) [reply src/main/java/woocommerce/utils/ReflectionUtils.java]: I agree.

File: src/main/java/woocommerce/entity/product/Product.java [hunk 2]
```diff
@@ -20,9 +20,9 @@
     private String slug;
     @JsonElement(key = "permalink")
     private String permalink;
-    @JsonElement(key = "date_created")
+    @JsonElement(key = "date_created", type = TypeOfJsonElement.ISO8601_DATETIME)
```
Comment (weirdobeardo48) [inline src/main/java/woocommerce/entity/product/Product.java:23]: It's good that we define this clear, but should we add one more property like "format" or something, so we would have 1 Date type and multiple formats, that would be more convenient than remembering the names.

**Comment by weirdobeardo48** [comment]
Hey @anhoangphuc,
Please create another pull request related to this to apply CI check. I will close this pull request for now!
Thanks!

**Pull request closed without merging**